packages/grid/x-data-grid/src/hooks/features/columnGrouping/useGridColumnGrouping.ts

```diff
@@ -0,0 +1,183 @@
+ */
+export const unwrapGroupingColumnModel = (
+  columnGroupingModel?: GridColumnGroupingModel,
+): UnwrappedGroupingModel => {
+  if (!columnGroupingModel) {
+    return {};
+  }
+
+  const rep: UnwrappedGroupingModel = {};
+
+  columnGroupingModel.forEach((columnGroupNode) => {
+    const unwrappedSubTree = recurrentUnwrapGroupingColumnModel(columnGroupNode, []);
+    Object.entries(unwrappedSubTree).forEach(([key, value]) => {
+      if (rep[key] !== undefined) {
+        throw new Error(
+          [
+            `MUI: columnGroupingModel has duplicated field.`,
+            `column field ${key} occurres two times in the grouping model:`,
+            `- ${rep[key].join(' > ')}`,
+            `- ${value.join(' > ')}`,
+          ].join('\n'),
+        );
+      }
+      rep[key] = value;
+    });
+  });
+
+  return rep;
+};
+
+const createGroupLookup = (columnGroupingModel: GridColumnNode[]): GridColumnGroupLookup => {
+  let groupLookup: GridColumnGroupLookup = {};
+
+  columnGroupingModel.forEach((node) => {
+    if (isLeaf(node)) {
+      return;
+    }
+    const { groupId, children, ...other } = node;
+    if (!groupId) {
+      throw new Error(
+        'MUI: An element of the columnGroupingModel does not have either `field` or `groupId`',
+      );
+    }
+    if (!children) {
+      console.warn(`MUI: group groupId=${groupId} has no children. `);
+    }
+    const groupParam = { ...other, groupId };
+    const subTreeLookup = createGroupLookup(children);
+    if (subTreeLookup[groupId] !== undefined || groupLookup[groupId] !== undefined) {
+      throw new Error(
+        `MUI: The groupId ${groupId} is used multiple times in the columnGroupingModel`,
+      );
+    }
+    groupLookup = { ...groupLookup, ...subTreeLookup, [groupId]: groupParam };
+  });
+
+  return { ...groupLookup };
+};
+export const columnGroupsStateInitializer: GridStateInitializer<
+  Pick<DataGridProcessedProps, 'columnGroupingModel'>
+> = (state, props) => {
+  const groupLookup = createGroupLookup(props.columnGroupingModel ?? []);
+  return {
+    ...state,
+    columnGrouping: { lookup: groupLookup, groupCollapsedModel: {} },
+  };
+};
+
+/**
+ * @requires useGridColumns (method, event)
+ * @requires useGridParamsApi (method)
+ */
+export const useGridColumnGrouping = (
+  apiRef: React.MutableRefObject<GridApiCommunity>,
+  props: Pick<DataGridProcessedProps, 'columnGroupingModel' | 'experimentalFeatures'>,
+) => {
+  /**
+   * API METHODS
+   */
+  const getColumnGroupPath = React.useCallback<
+    GridColumnGroupingApi['unstable_getColumnGroupPath']
+  >(
+    (field) => {
+      const columnLookup = gridColumnLookupSelector(apiRef);
+
+      return columnLookup[field]?.groupPath ?? [];
+    },
+    [apiRef],
+  );
+
+  const getAllGroupDetails = React.useCallback<
+    GridColumnGroupingApi['unstable_getAllGroupDetails']
+  >(() => {
+    const columnGroupLookup = gridColumnGroupsLookupSelector(apiRef);
+    return columnGroupLookup;
```

Comment:
It felt more logical since all the other data are obtained with api getter

https://github.com/alexfauquette/mui-x/blob/columnGroup/packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx#L106:L106